### PR TITLE
Feature/48 frontend create resumes

### PR DESCRIPTION
**Pull request by lilian-delgado-jalafund**
[AB#48](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/48): Add create-resume routing
[AB#48](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/48): Create resume not yet tested on base project

[AB#48](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/48): Test API with Jsonplaceholder
I tested post with Jsonplaceholder, since backend file isn't finish (solution is not working). Commenting what should be the implemented code once backend is finished.
testjsonplaceholder

### CODE REVIEW

**Review by jorge-lema-jalafund — DISMISSED** [review]
You could have used the Yup library to create your validation schema. It's an improvement for next sprint!
Other than that it looks good

File: src/components/Resume/ResumeCreationApi.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import axios from "axios";
+
+export async function PostResume(resume) {
```
Comment (DaleneyRuben) [inline src/components/Resume/ResumeCreationApi.js]: Only fuctional components start with uppercase, **postResume** is the correct name

File: src/components/Resume/ResumeCreation.jsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    },
+    validate,
+    onSubmit: (values) => {
+      PostResume(values).then((result) => {
+        if (!result) {
+          history.push("/resumecreation");
+        } else {
+          alert("Success! Welcome to TeamUp!");
+          return result;
+        }
```
Comment (DaleneyRuben) [inline src/components/Resume/ResumeCreation.jsx]: Could you try changing this to async/await? It's better to use async/await ove promises

File: src/components/Resume/ResumeCreation.jsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  } else if (values.phone < 60000000 || values.phone >= 80000000) {
+    errors.phone = "Please enter a valid phone number with 8 characters";
+  }
+  if (!values.summary) {
+    errors.summary = "Bio description field is required";
+  } else if (values.summary.length > 150) {
+    errors.summary = "Please enter a description with 150 characters or less";
+  }
+  return errors;
+};
```
Comment (DaleneyRuben) [inline src/components/Resume/ResumeCreation.jsx]: With the yup library this can be done so easily

File: src/components/Resume/ResumeCreation.jsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+              name="firstname"
+              type="text"
+              error={formik.touched.firstname && formik.errors.firstname}
+              placeholder="Your name"
+              helperText={
+                formik.touched.firstname && formik.errors.firstname
+                  ? formik.errors.firstname
+                  : ""
+              }
+            ></TextField>
```
Comment (DaleneyRuben) [inline src/components/Resume/ResumeCreation.jsx]: you can close the component like this 

```
<TextField
   ...
   placeholder="Your name"
/>
```

File: src/components/Resume/ResumeCreation.jsx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            <TextField
+              onChange={formik.handleChange}
+              onBlur={formik.handleBlur}
+              value={formik.values.firstname}
+              id="firstname"
+              name="firstname"
+              type="text"
+              error={formik.touched.firstname && formik.errors.firstname}
+              placeholder="Your name"
+              helperText={
```
Comment (DaleneyRuben) [inline src/components/Resume/ResumeCreation.jsx]: can you order all this props alphabetically please? here and in all components you're passing props

**Review by DaleneyRuben — CHANGES_REQUESTED** [review]

File: src/components/App/App.jsx [hunk 6]
```diff
@@ -1,13 +1,35 @@
...
+          <Route exact path="/projectcreation"></Route>
+          <Route exact path="/projectdetail/:id"></Route>
+          <Route exact path="/resumelist"></Route>
+          <Route
+            className="layout-resume-creation"
+            exact
+            path="/resumecreation"
+            component={ResumeCreation}
+          ></Route>
+          <Route exact path="/resumedetail/:id"></Route>
```
Comment (DaleneyRuben) [inline src/components/App/App.jsx]: we need to think more about the naming of our routes. can you add a TODO comment?
```
// TODO revisit route names
```

**Review by DaleneyRuben — CHANGES_REQUESTED** [review]

File: src/components/Resume/ResumeCreationApi.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+
+export async function postResume(resume) {
+  try {
+    await axios.post("http://localhost:7071/api/resumes", resume);
```
Comment (DaleneyRuben) [inline src/components/Resume/ResumeCreationApi.js]: Check the axios [response schema](https://axios-http.com/docs/res_schema) you can use the `statusText`. If the backend returns an error you won't get it inside the `catch`, the backend must return an error message with a status code if something went wrong. Check [this](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) 
For instance:
```
{
   statusText: 'Forbidden',
   status: 403,
   data: {
      errorMessage: 'you do not have permissions'
   }
}
```

in this case although the post request failed you'll be returning a true.

**Review by DaleneyRuben — CHANGES_REQUESTED** [review]

File: src/components/Resume/ResumeCreation.jsx [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+                  : ""
+              }
+              id="summary"
+              maxRows={4}
+              multiline
+              name="summary"
+              onBlur={formik.handleBlur}
+              onChange={formik.handleChange}
+              placeholder="Enter a brief description of you. 
+              Describe your professionalskills and experience."
```
Comment (jorge-lema-jalafund) [inline src/components/Resume/ResumeCreation.jsx]: ** change professionalskills to professional skills please

**Review by jorge-lema-jalafund — CHANGES_REQUESTED** [review]

**Review by DaleneyRuben — DISMISSED** [review]
LGTM 👍🏻

File: src/components/Resume/ResumeCreation.jsx [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+  email: Yup.string()
+    .email("Invalid email address")
+    .required("Email field is required"),
+  firstName: Yup.string()
+    .min(3, "Name is too short - should be 3 characters minimum")
+    .matches(/^[A-Za-z ]*$/, "Please enter valid name")
+    .max(20, "Please enter a name with 20 characters or less")
+    .required("Name field is required"),
+  lastName: Yup.string()
+    .min(5, "Lastname is too short - should be 5 characters minimum")
```
Comment (paulo-teran-jalafund) [inline src/components/Resume/ResumeCreation.jsx]: Maybe a minimum of 5 is too restrictive, there are lastnames as "Lara" or "Cruz" or "Oña" with less than 5 characters

Comment (lilian-delgado-jalafund) [reply src/components/Resume/ResumeCreation.jsx]: Ok, noted 👍🏻

**Review by jorge-lema-jalafund — DISMISSED** [review]
looks good to me

File: src/components/Resume/ResumeCreation.jsx (hunk 9 above)
Comment (DaleneyRuben) [reply src/components/Resume/ResumeCreation.jsx]: Good catch @paulo-teran-jalafund

**Review by DaleneyRuben — COMMENTED** [review]
It's all looking good. Just one thing I can see that you're running push --force every time. Why?
Screenshot from 2021-07-22 09-57-31
Screenshot from 2021-07-22 09-57-22
Screenshot from 2021-07-22 09-57-10

**Comment by lilian-delgado-jalafund** [comment]
@DaleneyRuben  I’m using force push because I learned to use the rebase strategy...so every time I pull from origin/develop to sync my local branch (and get the group's last changes), I re-write them making a rebase and not with merge commits. However, it happened to me more in this repository since we're many people editing the pipeline file

**Review by ela-ocana-jalafund — DISMISSED** [review]

**Review by paulo-teran-jalafund — DISMISSED** [review]

**Comment by DaleneyRuben** [comment]
@lilian-delgado-jalafund merge conflicts here

**Review by paulo-teran-jalafund — APPROVED** [review]

**Review by jorge-lema-jalafund — APPROVED** [review]

**Review by ela-ocana-jalafund — APPROVED** [review]

**Pull request merged**